core/mempool/src/lib.rs

```diff
@@ -22,6 +22,7 @@ pub mod payload_gossip;
 use payload_gossip::PayloadGossip;
 
 const POISONED_LOCK_ERR: &str = "The lock was poisoned.";
+const BLOCK_SIZE: u32 = 100;
```

Comment:
At one block per second production it will limit the throughput of our system to 100tps. We should make it configurable and larger. First, we should place it into `Client` config, and then make it larger, e.g. 1000. The problem is that if someone wants to benchmark our code, they would have to pass a different threshold through a command line which currently requires rebuilding docker image, which is cumbersome.

Reply:
Making it larger sounds good, but I don't think this should be in client config because it is not something configurable but rather part of the protocol and should be the same for every full node.

Reply:
We need to allow it to be configurable for benchmarking and testing reasons. We will need to have regression tests that check that `BLOCK_SIZE` + delta works because we would want to have the default value of `BLOCK_SIZE` to be less than the point at which the system breaks. And benchmarks will have to be run for different values of this so that we can tune it every now and then.

Also, I wouldn't consider that to be the part of the protocol. It does actually have to be, because multiple nodes can attempt to exchange their payloads even if their `BLOCK_SIZE` is different, if some node decides to use a payload that is too large and cannot be fetched it will be automatically not voted for. We should keep minimal the number of variables that are considered to be fixed for the protocol.

Reply:
But it seems silly that consensus cannot be reached because different nodes have use different block sizes. It seems that making this configurable will incentivize people to use very small block size in order to avoid not being able to produce a block.

Reply:
We need a separate mechanism to prevent people from censoring every transaction and submitting an empty block anyway. That has to be an economic incentive. Having this parameter configurable through a command line oppose to a source code creates little additional incentive, especially for those who are savvy enough to look into it. Not making it configurable will however create difficulty for us in testing. We are going to extract other parameters into configs too, like gossip frequency for peer status and Nightshade. Block announcement and block fetching parameters. Until we get into a solid stable state we will be fiddling with these parameters a lot.

Reply:
Also cache sizes are going to be configurable, too.

Reply:
> But it seems silly that consensus cannot be reached because different nodes have use different block sizes. It seems that making this configurable will incentivize people to use very small block size in order to avoid not being able to produce a block.

Consensus will always be reached, because everyone will be able to vote on the smallest payload.